timesketch/frontend-ng/src/components/LeftPanel/SigmaRule.vue

```diff
@@ -92,21 +138,97 @@ export default {
       return this.$store.state.sketch
     },
     sigmaRuleSummary() {
-      const fields = ['author', 'description', 'references', 'date', 'modified', 'falsepositives', 'level', 'tags']
+      const fields = [
+        'author',
+        'description',
+        'references',
+        'status',
+        'date',
+        'modified',
+        'falsepositives',
+        'level',
+        'tags',
+        'rule_uuid',
+        'search_query',
+      ]
       return Object.fromEntries(Object.entries(this.sigmaRule).filter(([key]) => fields.includes(key)))
     },
   },
   methods: {
     search(queryString) {
+      // if sketch.id is undefined, we are in the sigma list view and not searching
+      if (!this.sketch.id) {
+        return
+      }
       let eventData = {}
       eventData.doSearch = true
       eventData.queryString = queryString
       eventData.queryFilter = defaultQueryFilter()
       console.log(eventData)
       EventBus.$emit('setQueryAndFilter', eventData)
     },
+    editSigmaRule(ruleUuid) {
+      console.log('Edit Rule: ' + ruleUuid)
+      this.$router.push({
+        name: 'Studio',
+        params: {
+          id: ruleUuid,
+          type: 'sigma',
+        },
+      })
+    },
+    deleteRule(ruleUuid) {
+      if (confirm('Delete Rule?')) {
+        ApiClient.deleteSigmaRule(ruleUuid)
+          .then((response) => {
+            console.log('Rule deleted: ' + ruleUuid)
+            this.$store.dispatch('updateSigmaList')
+            this.$router.push({
+              name: 'Studio',
+              params: {
+                id: 'new',
+                type: 'sigma',
+              },
+            })
+          })
+          .catch((e) => {
+            console.error(e)
+          })
+      }
+    },
+    deprecateSigmaRule(ruleUuid) {
+      // this method is not really deprecating the rule, but for lack of better
+      // words, it is changing the status to deprecated. This means the rule
+      // will not be picked up by the Sigma analyzer.
+      if (confirm('Deprecate Rule?')) {
+        //  get the current Sigma rule yaml again
+        ApiClient.getSigmaRuleResource(ruleUuid)
+          .then((response) => {
+            var editingRule = response.data.objects[0]
+            const regex = /status:\s*(experimental|testing|stable)/g
```

Comment:
hm in theory they can put in arbitrary code yes, for this functionality, it would only matter if it was "stable" before. Should I focus on stable therefor only?